### PR TITLE
PR: Improve dividers between cards on mobile

### PR DESCRIPTION
**Pull request by davemart-in**
## Description

This relates to https://github.com/Automattic/loop/issues/177

Dividers between cards could probably benefit from being 100% width and slightly darker, again to just help you see at a glance as you're flipping through your feed when you've moved from one card to the next:

## Before

Image

## After

CleanShot 2024-05-01 at 12 53 08@2x

### CODE REVIEW

**Comment by Addison-Stavlo** [comment]
At first glance im confused why we are changing 11 separate files to achieve this?

**Comment by Addison-Stavlo** [comment]
I ran into something weird I cant seem to repro elsewhere. If i go to conversations, collapse the browser width to mobile, and slowly widen it back out... the sidebar gets stuck in collapsed mode?

Screenshot 2024-05-03 at 8 27 20 AM

I wouldn't expect this looking at the changes but cant seem to repro on production. Are you seeing this as well?

**Comment by davemart-in** [comment]
> I wouldn't expect this looking at the changes but cant seem to repro on production. Are you seeing this as well?

I am. How weird is that?

Digging in...

**Comment by davemart-in** [comment]
@Addison-Stavlo though I'm not seeing this locally.

**Comment by davemart-in** [comment]
Also noticed that it catches up and resolves itself after like 5 seconds. That's weird. Looks like the `.is-global-sidebar-collapsed` class removal is being delayed for some reason?

**Comment by Addison-Stavlo** [comment]
> Also noticed that it catches up and resolves itself after like 5 seconds. That's weird. Looks like the .is-global-sidebar-collapsed class removal is being delayed for some reason?

If we look at the selector causing this in https://github.com/Automattic/wp-calypso/pull/90280 there are 2 issues. One is that it can return true outside of the sites dashboard sections (it can be true simply for its breakpoint and redesign flag). That PR is updating the check to ensure we are on the sites dashboard.

The second issue is the use of the `isWithinBreakpoint` function. Because this is simply the `isWithinBreakpoint` function and not a `useBreakpoint` hook within the component, the changing breakpoint will not cause the component to rerender and update. Instead, it doesn't recalculate until the component rerenders for a separate reason. That would be the delay you see I think.

We could resolve the second issue delay by updating the components that use this selector. Instead of relying on `isWithinBreakpoint` within the selector, we use `useBreakpoint` in the components that are running that selector instead. Although im not sure if the expanding/contracting viewport size immediate responsiveness is a common user flow to worry about, and once things are limited to the sites dashboard where this collapsed version works as expected the delay may not be a big deal. 🤔

**Comment by davemart-in** [comment]
Yeah, I suspect your PR resolves this enough for now.

**Comment by Addison-Stavlo** [comment]
Ok, back to the changes here 😅.

In general the changes look great!  There is one thing im noticing in the header for the search stream though.


BEFORE
Screenshot 2024-05-03 at 9 35 16 AM


AFTER
Screenshot 2024-05-03 at 9 35 12 AM

Notice both the position change of:
* The relevance/date tabs above the input
* The "sites" tab header below the input

File: client/reader/search-stream/index.jsx [hunk 1]
```diff
@@ -173,11 +173,7 @@ class SearchStream extends React.Component {
 		return (
 			<div>
 				<DocumentHead title={ documentTitle } />
-				<div
-					className="search-stream__fixed-area"
-					style={ { width: this.props.width } }
-					ref={ this.handleFixedAreaMounted }
-				>
```
Comment (Addison-Stavlo) [inline client/reader/search-stream/index.jsx:180]: Seeing how everything else seems to be restricted behind a breakpoint check, I assume this is what is causing the weirdness in the search stream. 🤔

**Comment by davemart-in** [comment]
Okay @Addison-Stavlo I think this is ready for another review.

File: client/reader/search-stream/index.jsx (hunk 1 above)
Comment (davemart-in) [reply client/reader/search-stream/index.jsx:180]: Thanks. Yeah, this was related.

**Comment by Addison-Stavlo** [comment]
Looking good. There is one other place I see a regression. If we look at the subscriptions tab on the following feed in mobile.

BEFORE
Screenshot 2024-05-03 at 1 03 40 PM


AFTER
Screenshot 2024-05-03 at 1 03 46 PM

**Comment by davemart-in** [comment]
Okay I think this one is ready for another review. Thanks!

**Review by Addison-Stavlo — COMMENTED** [review]
Looks good.

Unit tests are failing because of a typescript check (it isnt expecting your new 'style' object on props). Yay typescript. I pushed a change so hopefully that should be good.

**Review by Addison-Stavlo — APPROVED** [review]
✅ when tests pass

**Pull request merged**